### ISSUE TITLE
How to evaluate a trained model and print indicators？

### ISSUE THREAD

**Issue by Idol-Dou2021** [roboflow/rf-detr#1110]
Hello, I have trained the model now. How can I output the evaluation indicators of the test set? Is there a script？

**Comment by anatoly-ryabchenko** [comment]
Hi @Idol-Dou2021 
There's no built-in method, but you can use PyTorch Lighting (PTL) for this:

```
from rfdetr.config import RFDETRMediumConfig, TrainConfig   # match the variant you trained
from rfdetr.training import RFDETRModelModule, RFDETRDataModule, build_trainer

model_config = RFDETRMediumConfig(num_classes=NUM_CLASSES)
train_config = TrainConfig(dataset_dir="path/to/dataset", output_dir="output")

module     = RFDETRModelModule(model_config, train_config)
datamodule = RFDETRDataModule(model_config, train_config)
trainer    = build_trainer(train_config, model_config)

# evaluate on the test split and print COCO metrics
trainer.test(module, datamodule, ckpt_path="output/checkpoint_best_total.pth")
```

**Comment by Borda** [comment]
@anatoly-ryabchenko would you be interested in adding this API to the model?

**Comment by anatoly-ryabchenko** [comment]
> [@anatoly-ryabchenko](https://github.com/anatoly-ryabchenko) would you be interested in adding this API to the model?

sure thing! can start on it tomorrow

**Comment by Idol-Dou2021** [comment]
I have carried out index evaluation for two categories of detection tasks. The code is as follows:

```javascript
import torch
from rfdetr.config import RFDETRLargeConfig, TrainConfig   # match the variant you trained
from rfdetr.training import RFDETRModelModule, RFDETRDataModule, build_trainer

model_config = RFDETRLargeConfig(num_classes=2)
train_config = TrainConfig(dataset_dir="/mnt/sda/rdd/yolo_sliced_final_1280_1280_rfdetr", output_dir="output_eval")

module     = RFDETRModelModule(model_config, train_config)
datamodule = RFDETRDataModule(model_config, train_config)
trainer    = build_trainer(train_config, model_config)

# Manually load checkpoint weights (bypass PL's loop state restoration which causes KeyError)
ckpt_path = "/mnt/sda/rdd/nohup/rfdetr/checkpoint_best_total.pth"
checkpoint = torch.load(ckpt_path, map_location="cpu", weights_only=False)
if "state_dict" in checkpoint:
    module.load_state_dict(checkpoint["state_dict"])
elif "model_state_dict" in checkpoint:
    module.load_state_dict(checkpoint["model_state_dict"])
else:
    module.load_state_dict(checkpoint)

# evaluate on the test split and print COCO metrics
trainer.test(module, datamodule)

```
The operation results are as follows：

<img width="740" height="644" alt="Image" src="https://github.com/user-attachments/assets/0b271eda-9260-4bbd-b17f-2cd67559e526" />
